lib/miq_expression.rb

```diff
@@ -524,7 +524,22 @@ def self._to_ruby(exp, context_type, tz)
       operands = operands2rubyvalue(operator, exp[operator], context_type)
       clause = operands.join(" #{normalize_ruby_operator(operator)} ")
     when "contains"
-      operands = operands2rubyvalue(operator, exp[operator], context_type)
+      exp[operator]["tag"] ||= exp[operator]["field"]
+      operands = if context_type != "hash"
+                   ref, val = value2tag(preprocess_managed_tag(exp[operator]["tag"]), exp[operator]["value"])
+                   ["<exist ref=#{ref}>#{val}</exist>"]
+                 elsif context_type == "hash"
+                   # This is only for supporting reporting "display filters"
+                   # In the report object the tag value is actually the description and not the raw tag name.
+                   # So we have to trick it by replacing the value with the description.
+                   description = MiqExpression.get_entry_details(exp[operator]["tag"]).inject("") do|s, t|
```

Comment:
wow. I've read this code a few times and have no idea with this `inject` does.

BUT

this code is not changing it, so that is not important